src/super_gradients/training/transforms/pipeline_adaptors.py

```diff
@@ -62,11 +67,17 @@ def post_transforms_processing(self, sample):
                 sample["labels"] = np.zeros((0))
             if len(sample["is_crowd"]) == 0:
                 sample["is_crowd"] = np.zeros((0))
-            return DetectionSample(
+            sample = DetectionSample(
                 image=sample["image"],
                 bboxes_xyxy=np.array(sample["bboxes"]),
                 labels=np.array(sample["labels"]),
                 is_crowd=np.array(sample["is_crowd"]),
                 additional_samples=None,
             )
-        return sample["image"]
+        elif self.sample_type == SampleType.SEGMENTATION:
+            sample = SegmentationSample(image=Image.fromarray(sample["image"]), mask=Image.fromarray(sample["mask"]))
```

Comment:
The reason for the casting to PIL image is because this is what all of our segmentation transforms expect.
So if the next transform in the pipeline is a native one we have to cast it.
See also my next comment about how I see the future support for segmentation transforms.